pallets/stake/src/lib.rs

```diff
@@ -615,6 +620,20 @@ decl_module! {
 		/// Minimum stake for any registered on-chain account to become a nominator
 		const MinNominatorStk: BalanceOf<T> = T::MinNominatorStk::get();
 
+		#[weight = 0]
+		fn set_blocks_per_round(
+			origin,
+			blocks_per_round: u32,
+		) -> DispatchResult {
+			frame_system::ensure_root(origin)?;
+			ensure!(
+				blocks_per_round >= T::MinBlocksPerRound::get(),
+				Error::<T>::BlocksPerRoundBelowMin
+			);
+			<BlocksPerRound>::put(blocks_per_round);
+			Self::deposit_event(RawEvent::BlocksPerRoundSet(blocks_per_round));
+			Ok(())
+		}
```

Comment:
I think setting this immediately may be dangerous. Imagine it was 600 then we lower it to 17. The logic on-chain is to mod the block number by 17. It may be that the round changes immediately if we happen to be at a multiple of 17.

I think at minimum we should finish the already started round before changing. The maybe the logic for when the next round changes could be something like:

`next_round_starts_at = (current_block - last_round_ended_at) % current_round_length`

Reply:
Yeah, I think you're right. Bummer that making this configurable, adds so much overhead.

Reply:
For real, I just tried to code it up and now I'm thinking about how it effects the delayed payments. The current design doesn't snapshot issuance per round during the round, but instead calculates it when the rewards are paid out after a `RoundDuration` delay. So that needs to change to make sure it doesn't mess up issuance...

Reply:
I have a solution. Will push soon.

Reply:
Definitely, the scenario you described above is still not addressed by my proposed design. Here's an example I came up with:

The previous BlocksPerRound was 20 and the new BlocksPerRound is 11. The change is made between blocks 100-120 so at block 120, the new interval is set to 11 and the 7th round starts. In this case, the 7th round only lasts 1 block because 121 % 11 == 0.

Reply:
So the way I wrote it now, it won't switch from the 7th to 8th round until block 132 so it'll skip the first new round transition at 121. This means the duration of the round immediately after the round when the interval is changed is of length `[new + 1, 2*new]`. This is safer than it being of length `[1, new]` which is what it was before.